service/src/main/java/bio/terra/pipelines/db/PipelinesDao.java

```diff
@@ -17,11 +21,26 @@ public class PipelinesDao {
               rs.getString("display_name"),
               rs.getString("description"));
 
-  private final JdbcTemplate tpsJdbcTemplate;
+  private final NamedParameterJdbcTemplate jdbcTemplate;
 
   @Autowired
   public PipelinesDao(TspsDatabaseConfiguration tspsDatabaseConfiguration) {
-    this.tpsJdbcTemplate = new JdbcTemplate(tspsDatabaseConfiguration.getDataSource());
+    this.jdbcTemplate = new NamedParameterJdbcTemplate(tspsDatabaseConfiguration.getDataSource());
+  }
+
+  /**
+   * Return the matching pipeline, given the pipelineId and version
+   *
+   * @param pipelineId
+   * @return Pipeline object
+   */
+  public Pipeline getPipeline(String pipelineId) {
+    DbPipeline dbPipeline =
+        getDbPipelineIfExists(pipelineId)
+            .orElseThrow(
+                () -> new NotFoundException(String.format("Pipeline %s not found.", pipelineId)));
```

Comment:
check to see whether the common library calls NotFoundException elsewhere

Reply:
as discussed on Slack: terra-common-lib doesn't have other calls of NotFoundException, but other repos (e.g. WSM) do create custom classes for each [Object]NotFoundException, so we're opting to follow the apparent DSP convention and create a PipelineNotFoundException class